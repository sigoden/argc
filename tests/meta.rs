use crate::*;

#[test]
fn dotenv() {
    let script = r###"
# @meta dotenv
"###;
    snapshot!(script, &["prog"]);
}

#[test]
fn dotenv_custom_path() {
    let script = r###"
# @meta dotenv .env.local
"###;
    snapshot!(script, &["prog"]);
}
