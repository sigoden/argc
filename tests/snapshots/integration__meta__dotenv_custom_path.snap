---
source: tests/meta.rs
expression: data
---
RUN
prog

# OUTPUT
_argc_load_dotenv() {
    local env_file="$1" env_vars=""
    if [[ -f "$env_file" ]]; then
        while IFS='=' read -r key value; do
            if [[ "$key" == $'#'* ]] || [[ -z "$key" ]]; then
                continue
            fi
            if [[ -z "${!key+x}" ]]; then
                env_vars="$env_vars $key=$value"
            fi
        done < <(cat "$env_file"; echo "")
        if [[ -n "$env_vars" ]]; then
            eval "export $env_vars"
        fi
    fi
}
_argc_load_dotenv .env.local
argc__args=( prog )
argc__positionals=(  )

# BUILD_OUTPUT
argc__args=([0]="prog")
argc__positionals=()
